ansible.go

```diff
@@ -4,65 +4,105 @@ package main
 
 import (
 	"encoding/json"
-	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
-	"github.com/spf13/viper"
+	"net"
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
 )
 
 // AnsibleConfig is a collection of meta-information about an Ansible repository.
 //
 // The virtualenv specified by the VenvConfig needs to be initialized before running Ansible commands.
 // All dir paths are relative to the root of the source tarball.
 type AnsibleConfig struct {
-	VenvConfig         VenvConfig // Virtualenv config that Ansible will be executed in
-	Cwd                string     // Path to change to when running Ansible commands
-	InventoryList      []string   // Paths to all desired inventories
+	VenvConfig    VenvConfig // Virtualenv config that Ansible will be executed in
+	Cwd           string     // Path to change to when running Ansible commands
+	InventoryList []string   // Paths to all desired inventories
+}
+
+// CreateAnsibleTargetsList generates and returns an array of possible targets
+// The possible targets are ip addresse from the host interfaces or the hostname itself
+func CreateAnsibleTargetsList() []string {
+	var inventoryTargets = []string{}
+
+	// extract all ip addresses except the loopback one and add the hostname as a last fallback mechanism
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		logrus.Fatal("Unable to get the network interfaces")
```

Comment:
We typically don't want to use Fatal logging, and instead return an error.
```suggestion
		return []string{}, errors.Wrap(err, "Unable to get the network interfaces")
```